src/Microsoft.TestPlatform.Common/Filtering/FilterHelpers.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.Common.Filtering
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Linq;
+    using System.Text;
+    using Microsoft.VisualStudio.TestPlatform.Common.Resources;
+
+    public static class FilterHelpers
+    {
+        private const char EscapePrefix = '%';  
```

Comment:
Can we  use `\`(backslash) as escape character convert ( `(` -> `\(`, `&` -> `\&`,etc..) and write the our own parsing logic rather than depending upon regex? To make easy to construct the filter from CLI?

Reply:
I agree with you that this is hard to use in CLI scenario. As you can see from the out-of-sync comment you pointed out below, I actually started with exactly what you suggested, but quickly realized that it would involve a bigger change, which I was reluctant to do this late and I really want this fix for preview 3, as it's a serious regression for LUT. Let me give this a try today and see how it goes.

Reply:
Done.